ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/router_bgp.py

```diff
@@ -196,7 +196,14 @@ def _router_bgp_vrfs(self) -> list | None:
                             "prefix_list_out": bgp_peer.pop("prefix_list_out", None),
                         }
                     )
-                    bgp_vrf.setdefault(address_family, {}).setdefault("neighbors", []).append(neighbor)
+
+                    append_if_not_duplicate(
+                        list_of_dicts=bgp_vrf.setdefault(address_family, {}).setdefault("neighbors", []),
+                        primary_key="ip_address",
+                        new_dict=neighbor,
+                        context="BGP peer defined under VRFs",
+                        context_keys=["ip_address"],
+                    )
```

Comment:
Can you add a negative unit test for this? 
like a duplicate for the same host with different configs? 
Seems important that we don't lose track of this.

Reply:
Added.